var/spack/repos/builtin/packages/gsl/package.py

```diff
@@ -24,3 +24,31 @@ class Gsl(AutotoolsPackage, GNUMirrorPackage):
     version('2.1',   sha256='59ad06837397617f698975c494fe7b2b698739a59e2fcf830b776428938a0c66')
     version('2.0',   sha256='e361f0b19199b5e6c21922e9f16adf7eca8dd860842802424906d0f83485ca2d')
     version('1.16',  sha256='73bc2f51b90d2a780e6d266d43e487b3dbd78945dd0b04b14ca5980fe28d2f53')
+
+    variant('external-cblas', default=False, description='Build against external blas')
+
+    # from https://dev.gentoo.org/~mgorny/dist/gsl-2.3-cblas.patch.bz2
+    patch('gsl-2.3-cblas.patch', when="+external-cblas")
+
+    conflicts('+external-cblas', when="@:2.2.9999")
+    depends_on('m4',       type='build', when='+external-cblas')
+    depends_on('autoconf', type='build', when='+external-cblas')
+    depends_on('automake', type='build', when='+external-cblas')
+    depends_on('libtool',  type='build', when='+external-cblas')
+    depends_on('blas', when='+external-cblas')
+
+    @property
+    def force_autoreconf(self):
+        # The external cblas patch touches configure
+        return self.spec.satisfies('+external-cblas')
```

Comment:
Alternatively, we could use the patch to regenerate configure, then develop our own patch to configure. That way we don't need to build the extra dependencies. Up to you.

Reply:
I usually don’t like that approach as patches against configure quickly become very messy and big if you use exactly the same autotools, plus we might end up with different patches for each version of gsl.